src/protocols/ProtocolEthernet.cpp

```diff
@@ -0,0 +1,60 @@
+#include "ProtocolEthernet.h"
+#include <unistd.h>
+#include <sstream>
+
+namespace RACprotocol
+{
+    ProtocolEthernet::ProtocolEthernet() : pProtoStruct(nullptr)
+    {
+	pProtoStruct = new protocol;
+	bzero(pProtoStruct, sizeof(*pProtoStruct));
+    }
+
+    ProtocolEthernet::~ProtocolEthernet()
+    {
+	if (pProtoStruct != nullptr)
+	{
+	    delete pProtoStruct;
+	}
+    }
+
+    std::string		ProtocolEthernet::getMACFormated(const uint8_t *uMAC) const
+    {
+	char	cMAC[64];
+
+	bzero(cMAC, 64);
+	sprintf(cMAC, "%02x::%02x::%02x::%02x::%02x::%02x", 
+		(int)uMAC[0], (int)uMAC[1], (int)uMAC[2], (int)uMAC[3], (int)uMAC[4], (int)uMAC[5]);
+
+	return std::string(cMAC);
+    }
+
+    const std::string	ProtocolEthernet::derivedGetProtocolFormated() const 
+    {
+	std::stringstream ss;
+
+	ss << std::endl << "Src MAC:" << getMACFormated(pProtoStruct->uSrcMAC) << std::endl << "Dest MAC:" << getMACFormated(pProtoStruct->uDestMAC) << std::endl << "EtherType: " << getEtherType() << std::endl;
+
+	return ss.str();
+    }
+
+    void	ProtocolEthernet::derivedSetStructProtocol(const char* data)
```

Comment:
Here you pass a `data` but do not specify any size of it. You won't overflow your `pProtoStruct` buffer since you only read `sizeof(pProtoStruct)` but you won't easyly know if you have a bug here (like your `data` is important and bigger than the `pProtoStruct` and you end up with corrupted data)

Reply:
The size of my data is often bigger than the size of the pProtoStruct seeing that the data correspond at the whole protocol that we receive.
I only treat untill as the size of my pProtoStruct and i ignore the rest for this "turn". ( if i detect an knonw protocol i send data + sizeof(pProtoStruct) at the next function that will treat the data and so on)
If the size data is smaller that my pProtoStruct seing that i filled my pProtoStruct of 0 i will do nothing (and obviously i will detect an unknow protocol and just display what i have been able to decode).
Maybe in the futur i will add a feature to detect a BAD FORMAT of a specifique PACKET PROTOCOL. (by checking the size of something else, but not now)